tests/drivers/adc/adc_rescale/src/main.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2023 The ChromiumOS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/adc/adc_emul.h>
+#include <zephyr/drivers/adc/adc_voltage_divider.h>
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+
+/* Raw to millivolt conversion doesn't handle rounding */
+#define MV_OUTPUT_EPS 10
+
+#define ADC_TEST_NODE_0 DT_NODELABEL(sensor0)
+
+/**
+ * @brief Get ADC emulated device
+ *
+ * @return pointer to ADC device
+ */
+const struct device *get_adc_device(void)
+{
+	const struct device *const adc_dev = DEVICE_DT_GET(DT_NODELABEL(adc0));
+
+	zassert_true(device_is_ready(adc_dev), "ADC device is not ready");
+
+	return adc_dev;
+}
+
+ZTEST_USER(adc_rescale, test_get_adc_device)
+{
+	zassert_true(get_adc_device());
```

Comment:
true? returns a struct device?

Reply:
Originally, I wanted to make sure `get_adc_device()` returns a real pointer and is not `NULL`, but looking at it again, the assert `zassert_true(device_is_ready(adc_dev), "ADC device is not ready");` in `get_adc_device()` already does that so the check is not necessary. I have removed it.